src/wp-includes/option.php

```diff
@@ -364,6 +364,172 @@ function get_options( $options ) {
 	return $result;
 }
 
+/**
+ * Sets the autoload values for multiple options in the database.
+ *
+ * Autoloading too many options can lead to performance problems, especially if the options are not frequently used.
+ * This function allows modifying the autoload value for multiple options without changing the actual option value.
+ * This is for example recommended for plugin activation and deactivation hooks, to ensure any options exclusively used
+ * by the plugin which are generally autoloaded can be set to not autoload when the plugin is inactive.
+ *
+ * @since 6.4.0
+ *
+ * @global wpdb $wpdb WordPress database abstraction object.
+ *
+ * @param array $options Associative array of option names and their autoload values to set. The option names are
+ *                       expected to not be SQL-escaped. The autoload values accept 'yes'|true to enable or 'no'|false
+ *                       to disable.
+ * @return array Associative array of all provided $options as keys and boolean values for whether their autoload value
+ *               was updated.
+ */
+function wp_set_option_autoload_values( array $options ) {
+	global $wpdb;
+
+	if ( ! $options ) {
+		return array();
+	}
+
+	$grouped_options = array(
+		'yes' => array(),
+		'no'  => array(),
+	);
+	$results         = array();
+	foreach ( $options as $option => $autoload ) {
+		wp_protect_special_option( $option ); // Ensure only valid options can be passed.
+		if ( 'no' === $autoload || false === $autoload ) { // Sanitize autoload value and categorize accordingly.
+			$grouped_options['no'][] = $option;
+		} else {
+			$grouped_options['yes'][] = $option;
```

Comment:
The documentation says that `'no'|false` and `'yes'|true` are accepted values. I'm wondering if we should be specific here, and either ignore or reject other values to ensure it's only ever used correctly. A docs change to explicitly open this up to `'no'|false to disable, anything else to enable` may occur in months/years to come if _misusage_ becomes prevalent enough. Being strict now would help prevent us needing to be loose forever.

Could be straightforward like:
```php
} elseif ( 'yes' === $autoload || true === $autoload ) {
```

An `else` block to notify of an unaccepted value might be good so it doesn't fail silently.

Reply:
I don't think that's necessary, it would be misaligned with how the other existing core functions treat these values. Core is generally very forgiving, and especially since these are effectively booleans, it's not crucial to be that strict.